L1Trigger/DTTrackFinder/src/L1MuDTTrackFinder.cc

```diff
@@ -181,36 +158,31 @@ void L1MuDTTrackFinder::run(const edm::Event& e, const edm::EventSetup& c) {
     reset();
 
     // run sector processors
-    L1MuDTSecProcMap::SPmap_iter it_sp = m_spmap->begin();
-    while (it_sp != m_spmap->end()) {
+    for (auto& sp : *m_spmap) {
       if (m_config->Debug(2))
-        cout << "running " << (*it_sp).second->id() << endl;
-      if ((*it_sp).second)
-        (*it_sp).second->run(bx, e, c);
-      if (m_config->Debug(2) && (*it_sp).second)
-        (*it_sp).second->print();
```

Comment:
This pattern of
```cpp
if (debug)
  // access ptr
if (ptr)
  // access ptr
if (debug && ptr)
  // access ptr
```
seems to repeat in this file many times. Given how the vectors and the map are filled, there should not be any `nullptr` entries, and therefore the `if (ptr)` check seems redundant. Also the nullness is not checked in the first `if`, so the code would (likely) crash anyway if, for any reason, the `ptr` would be null. Could this pattern be replaced with just
```cpp
if (debug)
  // access ptr
// access ptr
if (debug)
  // access ptr
```
?